flask/app.py

```diff
@@ -929,7 +929,11 @@ def run(self, host=None, port=None, debug=None,
             sn_host, _, sn_port = server_name.partition(':')
 
         host = host or sn_host or _host
-        port = int(port or sn_port or _port)
+
+        for possible_port_value in [port, sn_port, _port]:
```

Comment:
This could be simplified with something like this:

    port = next((int(x) for x in (port, sn_port, _port) if x is not None), None)

Not sure about the default value if all three are None.

Reply:
thanks for the suggestion @ThiefMaster ! I was considering doing a one-liner but decided not to, mainly because I was worried about readability... But your version doesn't looks pretty nice, definitely worth doing this for getting rid of 3 LOC, I'll amend it in a sec. I keep forgetting that `next` has this default value parameter, that's super useful.

As for default value, let's probably stick to how the old code handled it - `int(_port)` was the default if both `port` and `sn_port` were false-y.